packit/patches.py

```diff
@@ -26,17 +26,73 @@
 import logging
 from itertools import islice
 from pathlib import Path
-from typing import List, Tuple, Optional
+from typing import List, Optional
 
 import git
+import yaml
 
 from packit.constants import DATETIME_FORMAT
+from packit.git_utils import get_metadata_from_message
 from packit.local_project import LocalProject
 from packit.utils import is_a_git_ref, run_command
 
 logger = logging.getLogger(__name__)
 
 
+class PatchMetadata:
+    def __init__(
+        self,
+        name: Optional[str] = None,
+        path: Optional[Path] = None,
+        location_in_specfile: Optional[str] = None,
+        description: Optional[str] = None,
+        commit: Optional[git.Commit] = None,
+    ) -> None:
+        self.name = name
+        self.path = path
+        self.location_in_specfile = location_in_specfile
```

Comment:
I hope that we won't need to touch spec file with this metadata - but it's still good forward thinking to record such info

Reply:
Isn't this the solution for conditional patches? I thought we want to use that to uncomment the original part of spec-file.

But yes, I remember that we also discuss solution where we will not touch spec-file in that case.

What do you think about this? Is that what you want?
- Do not comment the line in spec and add e.g. `downstream:true` to the metadata when creating source-git.
- When we find `downstream:true` patch, we will only create a patch file and do not touch spec-file.

(Or we can treat commits with `patch_name` in metadata as downstream ones. I am not sure about this. People might want to enhance source-git patches as well.)

Reply:
As I am thinking about that. It would be much easier to make it in that way. If we don't touch spec-file, we can't do bad changes there..;)